aws/resource_aws_media_connect_flow.go

```diff
@@ -0,0 +1,526 @@
+						},
+						"smoothing_latency": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"stream_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ConflictsWith: []string{
+								"source.entitlement_arn",
+							},
+						},
+						"whitelist_cidr": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ConflictsWith: []string{
+								"source.entitlement_arn",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAWSMediaConnectFlowCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediaconnectconn
+
+	createOpts := &mediaconnect.CreateFlowInput{
+		Name: aws.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("availability_zone"); ok {
+		createOpts.AvailabilityZone = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("source"); ok {
+		createOpts.Source = expandAWSMediaConnectFlowSource(v.([]interface{}))
+	}
+
+	log.Printf("[DEBUG] Media Connect Flow create configuration: %#v", createOpts)
+	resp, err := conn.CreateFlow(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating elemental media connect flow: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.Flow.FlowArn))
+
+	return resourceAWSMediaConnectFlowRead(d, meta)
+}
+
+func resourceAWSMediaConnectFlowRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediaconnectconn
+
+	descOpts := &mediaconnect.DescribeFlowInput{
+		FlowArn: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Media Connect Flow describe configuration: %#v", descOpts)
+	resp, err := conn.DescribeFlow(descOpts)
+	if err != nil {
+		if isAWSErr(err, mediaconnect.ErrCodeNotFoundException, "") {
+			log.Printf("[WARN] Media Connect Flow %s not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error describing Media Connect Flow %s: %s", d.Id(), err)
+	}
+
+	d.Set("arn", resp.Flow.FlowArn)
+	d.Set("name", resp.Flow.Name)
+	d.Set("description", resp.Flow.Description)
+	d.Set("availability_zone", resp.Flow.AvailabilityZone)
+	d.Set("egress_ip", aws.StringValue(resp.Flow.EgressIp))
+	d.Set("source", flattenAWSMediaConnectFlowSource(resp.Flow.Source))
+
+	return nil
+}
+
+func resourceAWSMediaConnectFlowUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediaconnectconn
+
+	if d.HasChange("source") {
+		err := updateAWSMediaConnectFlowSource(conn, d.Id(), d.Get("source").([]interface{}))
+		if err != nil {
+			if isAWSErr(err, mediaconnect.ErrCodeNotFoundException, "") {
+				log.Printf("[WARN] Media Connect Flow %s not found, removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+			return fmt.Errorf("Error updating Media Connect Flow(%s) Source: %s", d.Id(), err)
+		}
+	}
+
+	stateConf := resource.StateChangeConf{
+		Pending: []string{
+			"UPDATING",
+			"STARTING",
+			"STOPPING",
+		},
+		Target: []string{
+			"STANDBY",
+			"ACTIVE",
+		},
```

Comment:
SDK constants available:

```suggestion
		Pending: []string{
			mediaconnect.StatusStarting,
			mediaconnect.StatusStopping,
			mediaconnect.StatusUpdating,
		},
		Target: []string{
			mediaconnect.StatusActive,
			mediaconnect.StatusStandby,
		},
```